contentcuration/contentcuration/frontend/channelList/views/Channel/ChannelDetailsModal.vue

```diff
@@ -0,0 +1,268 @@
+
+            <DetailsRow
+              v-if="channel.language"
+              :label="$tr('primaryLanguageHeading')"
+              :text="translateLanguage(channel.language)"
+            />
+            <Details :nodeID="channel.root_id" />
+          </div>
+        </VCardText>
+      </div>
+    </VCard>
+  </VDialog>
+
+</template>
+
+<script>
+
+  import { mapActions, mapGetters } from 'vuex';
+  import Vibrant from 'node-vibrant';
+  import Details from '../../../preview/views/Details';
+  import DetailsRow from '../../../preview/views/DetailsRow';
+  import { fileSizeMixin, constantsTranslationMixin } from 'shared/mixins';
+  import LoadingText from 'shared/views/LoadingText';
+  import CopyToken from 'shared/views/CopyToken';
+
+  export default {
+    name: 'ChannelDetailsModal',
+    components: {
+      Details,
+      LoadingText,
+      DetailsRow,
+      CopyToken,
+    },
+    mixins: [fileSizeMixin, constantsTranslationMixin],
+    props: {
+      channelId: {
+        type: String,
+      },
+    },
+    data() {
+      return {
+        loading: true,
+        loadError: false,
+        dominantColor: '',
+      };
+    },
+    computed: {
+      ...mapGetters('channelList', ['getChannel']),
+      channel() {
+        return this.getChannel(this.channelId);
+      },
+      thumbnail() {
+        return this.channel.thumbnail_url || '/static/img/kolibri_placeholder.png';
+      },
+      publishedDate() {
+        return this.$formatDate(this.channel.last_published, {
+          year: 'numeric',
+          month: 'long',
+          day: 'numeric',
+        });
+      },
+      backLink() {
+        return {
+          name: this.$route.matched[0].name,
+          query: this.$route.query,
+        };
+      },
+      routeParamID() {
+        return this.$route.params.channelId;
+      },
+      downloadOptions() {
+        return [
+          {
+            title: this.$tr('downloadCSV'),
+            href: window.Urls.get_channel_details_csv_endpoint(this.channel.id),
+          },
+          {
+            title: this.$tr('downloadPDF'),
+            href: window.Urls.get_channel_details_pdf_endpoint(this.channel.id) + '?condensed=true',
+          },
+          {
+            title: this.$tr('downloadDetailedPDF'),
+            href: window.Urls.get_channel_details_pdf_endpoint(this.channel.id),
+          },
+          {
+            title: this.$tr('downloadPPT'),
+            href: window.Urls.get_channel_details_ppt_endpoint(this.channel.id),
+          },
+        ];
+      },
+    },
+    watch: {
+      routeParamID(val) {
+        this.hideHTMLScroll(!!val);
+      },
+    },
+    mounted() {
+      this.load();
+      this.hideHTMLScroll(true);
+    },
+    methods: {
+      ...mapActions('channelList', ['loadChannel']),
+      load() {
+        this.loading = true;
+        this.loadChannel(this.channelId).then(() => {
+          this.loading = false;
+          let v = new Vibrant(this.thumbnail);
```

Comment:
Yes, pretty handy library too https://jariz.github.io/vibrant.js/

Reply:
That's really cool - I could imagine using it as an option for abstract pixel art thumbnails of topics with lots of descendants (rather than just mashing up their thumbnails).

Reply:
That would be really cool!